### PR TITLE
docs: mention python in debugging.md

### PR DESCRIPTION
**Pull request by psarna**
Evaluating Python code from within gdb is priceless,
especially that all helper classes and functions sourced from
scylla-gdb.py can be used in there. This commit adds a paragraph
in debugging.md mentioning this tool.

### CODE REVIEW

File: docs/guides/debugging.md [hunk 1]
```diff
@@ -922,3 +922,33 @@ latter, the following commands might be of help:
...
+GDB is well integrated with Python, and besides using the predefined set of commands from `scylla-gdb.py`, you can also run an interactive prompt by typing `python-interactive` or `pi` in gdb console. That allows you to benefit from the rich set of helper classes and wrappers defined in `scylla-gdb.py`, e.g. in order to iterate over one of the supported collection types (vectors, small vectors, maps, tuples, etc.) and print only the interesting bits.
+
+Example:
+```gdb
+(gdb) source /path/to/scylla-gdb.py
+(gdb) python-interactive
+>>> db = gdb.parse_and_eval("*(seastar::sharded<replica::database>*)debug::the_database")
+>>> instances = std_vector(db["_instances"])
+>>> for i, instance in enumerate(instances):
+...     print(i, instance["service"])
```
Comment (denesb) [inline docs/guides/debugging.md:936]: Unrelated note: we have a `$sharded_local()` convenience function just for this ;)

File: docs/guides/debugging.md [hunk 2]
```diff
@@ -922,3 +922,33 @@ latter, the following commands might be of help:
     scylla generate-object-graph # generates and visualizes an object graph
 
 Good luck, you are off the charted path here.
+
+#### Use python for inspecting objects and performing computations
+GDB is well integrated with Python, and besides using the predefined set of commands from `scylla-gdb.py`, you can also run an interactive prompt by typing `python-interactive` or `pi` in gdb console. That allows you to benefit from the rich set of helper classes and wrappers defined in `scylla-gdb.py`, e.g. in order to iterate over one of the supported collection types (vectors, small vectors, maps, tuples, etc.) and print only the interesting bits.
```
Comment (denesb) [inline docs/guides/debugging.md:927]: This file uses line length of 80. I know it is minor thing but please break these lines.
Maybe we can also mention how to transfer variables `python<->gdb`. One way is `gdb.parse_and_eval()` but this works one way and is sometimes award to use. There is a better way, using `gdb.convenience_variable()` and `gdb.set_convenience_variable()`.

**Comment by psarna** [comment]
* adhered to 80 lines in the paragraph
* dropped the ` ```gdb ` coloring specification because CI screamed at me that it's not a known extension (shame on them)

File: docs/guides/debugging.md (hunk 2 above)
Comment (denesb) [reply docs/guides/debugging.md]: Example:
```
(gdb) p debug::the_database
$1 = (seastar::sharded<replica::database> *) 0x7fffffffbdc0
(gdb) set $db=debug::the_database
(gdb) pi
>>> local_db = sharded(gdb.convenience_variable('db')).local()
>>> gdb.set_convenience_variable('local_db', local_db)
>>> 
(gdb) p $local_db
$2 = (replica::database *) 0x60000575e010
```

**Comment by psarna** [comment]
v3:
 - added a paragraph on convenience variables, courtesy of @denesb

**Pull request closed without merging**